reporter/src/main/kotlin/StatsCalculator.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model.utils
+
+import com.here.ort.model.DependencyTreeStats
+import com.here.ort.model.IssueStats
+import com.here.ort.model.OrtResult
+import com.here.ort.model.Severity
+import com.here.ort.model.Stats
+import com.here.ort.reporter.ResolutionProvider
+
+/**
+ * This class calculates [Stats] for a given [OrtResult] and applicable [com.here.ort.model.config.ErrorResolution]s
+ * and applicable [com.here.ort.model.config.RuleViolationResolution]s.
+ */
+internal class StatsCalculator {
```

Comment:
What about making `ortResult` and `resolutionProvider` class properties? Then they do not have to be passed around in the functions below.